src/components/select/select.scss

```diff
@@ -198,3 +196,5 @@ select.limel-select__native-control {
         margin-left: functions.pxToRem(-8);
     }
 }
+
+@import './partial-styles/_readonly';
```

Comment:
I think it depends on what you are importing.

The general rule is that if you import something that is supposed to be used in your styles, YES it should 100% be on top of the file. Let's say you import a bunch of mixins or SASS variables. If they are not first in the file, then rest of your SCSS code cannot use them properly and you will get either build errors or weird styles.

Note that we stopped using `import` for such cases, and instead use the `use`, because it has some other benefits. And they are still on top. [More here…](https://css-tricks.com/introducing-sass-modules/)

But this particular case, we import some partial styles here. Those are usually styles that may override the main styles. If we put them below the main styles, we have a lower risk that the overrides stop working, or we require less specifications in the partial styles. However, if the CSS is written with care, there shouldn’t be so much to worry about anyway. So maybe it is not a big deal.

But logically, you start specifying the **main styles** and **top-level styles** of something first, then you start adding more specific styles below. So it makes sense to have the partial styles written below the general styles; and in this case, "**imported**" below the general styles. Right? It's just the hierarchy of the styles.

I personally don't care that much where they are. But when we upgraded the MDC this summer, I noticed that @adrianschmidt moved the imports to the end of some files. Maybe it was an accident or maybe he had better reasons than what I explained above. 😊 

So as long as we decide for a convention, I'm fine with whichever.

Reply:
Huh, so I started moving imports to the end? I had totally forgotten that 😄

But I mostly agree with Kia's comment above. But I think context is more important than dogma here. As Kia mentions, if an imported file contains variables or mixins used in the current file, then yes, that import obviously has to come before those variables or mixins can be used.

When it comes to partial styles, I think it depends on what those partials include. What Kia said here makes sense:

> But logically, you start specifying the main styles and top-level styles of something first (like how the top level container looks like, or how the host element is), then you start adding more specific styles below.

From that follows that if you have general styles written in `my-component.scss` and then import more specific styles, that are really just broken out to a separate file in order to make the individual files smaller and group the styles a bit, it makes sense to do those imports at the end of the file. That's the case for this import. On the other hand, if we had a partial that defined a "base style", and imported that into several components, it would again make sense to put that import on top, as we would then add more specific styles in `my-component.scss`. Although, if we put it on top, we'll probably use `@use` rather than `@import`.

> I would not put the @imports to the bottom, you can check the convention about it here: [w3schools.com/cssref/pr_import_rule.asp](https://www.w3schools.com/cssref/pr_import_rule.asp)

First of all, never trust anything w3schools is saying about anything. People tend to believe w3schools is related to W3C (the World Wide Web Consortium) because of the similar name, but w3schools is a for-profit company that are more concerned about selling their online courses than about keeping their material up to date and correct 😞

I always use MDN for reference, instead of w3schools. Usually, if w3schools is the first search result you get, MDN will be one of the results close to the top. If you don't find results from MDN, just add `mdn` to whatever you searched for, and you will get MDN in your top hits. Here's the corresponding page from MDN: https://developer.mozilla.org/en-US/docs/Web/CSS/@import

Secondly, that page is talking about the use of `@import` in _css_ files. We're using `@import` in _scss_ files, and the `@import` statements are compiled out of the code when compiled to css, so the rules and limitations for `@import` in css doesn't apply here 🙂